packages/core/contracts/financial-templates/common/FundingRateApplier.sol

```diff
@@ -217,18 +217,17 @@ abstract contract FundingRateApplier is FeePayer {
         uint256 proposalTime = fundingRate.proposalTime;
         if (proposalTime != 0) {
             // Attempt to update the funding rate.
-            OptimisticOracle optimisticOracle = _getOptimisticOracle();
+            OptimisticOracleInterface optimisticOracle = _getOptimisticOracle();
             bytes32 identifier = fundingRate.identifier;
             bytes memory ancillaryData = _getAncillaryData();
 
             // Try to get the price from the optimistic oracle.
+            OptimisticOracleInterface.Request memory request =
```

Comment:
@mrice32 We can refactor this `getRequest` call above this try-catch since its called no matter what with the same params. This saves 100 bytes of bytecode

Reply:
@nicholaspai I worry about edge cases here because `getPrice` changes the state of the request, which could change the result of this call. This may be a non-issue, but it might be worth thinking through the implications.

Reply:
Hm this is a really good point. It's not an issue right now but I could imagine upgrading the OO in the future and failing to take this state modification into account, because `getPrice` is confusingly NOT a read only call. I moved the `getRequest` call into both the body of the try and the catch

Reply:
Sounds good!